pipeline/test/services/workload-cluster/testHNC.sh

```diff
@@ -42,8 +42,10 @@ function check_wc_hnc_creation_removal() {
     no_error=true
     debug_msg=""
 
-    mapfile -t user_namespaces < <(yq4 -e '.user.namespaces[]' "${config['config_file_wc']}")
-    mapfile -t user_admin_users < <(yq4 -e '.user.adminUsers[]' "${config['config_file_wc']}")
+    # shellcheck disable=SC2016
+    mapfile -t user_namespaces < <(yq4 -e '(.user.constraints | keys | .[]) as $nonHNCNamespaces | .user.namespaces[] | select(. != $nonHNCNamespaces)' "${config['config_file_wc']}")
```

Comment:
Have you tested this with multiple namespaces under constraints?

Not sure if our version supports this but `.user.namespaces - (.user.constraints | keys) | .[]` _should_ give you what you need.

Reply:
I tried your version with multiple namespaces under constraints and it worked well.
Thanks for the suggestion. It does feel like there is always a simpler way to write yq commands.